src/backend/google-auth.js

```diff
@@ -0,0 +1,156 @@
+      throw {
+        status: 403,
+        msg: `User email ${email} is not a member of the Leads group`,
+      };
+    }
+
+    // get user info
+    const userQueryResp = await db.users.getById(userId)
+
+    if (userQueryResp === -1) {
+      // No user found so make one
+      await db.users.createUser(email, given_name, family_name, userId)
+      await db.users.getById(userId)
+    }
+
+    // Check that the access token was sent along in the request
+    const authHeader = req.get("Authorization");
+    if (!authHeader) {
+      res.status(401).set('WWW-Authenticate', 'Bearer').send('No auth header found.').end();
+      return;
+    }
+    const [type, accessToken] = authHeader.split(' ');
+    if (type !== "Bearer" || !accessToken) {
+      res.status(401).set('WWW-Authenticate', 'Bearer').send('No bearer token found.').end();
+      return;
+    }
+    // Check whether the user has admin priveleges (admins can edit content using the CMS)
+    const resp = await db.featurePermissions.getAllowedActions(userId, accessToken)
+    console.log(resp);
+    const {allowedActions, groupIds} = resp;
+    if (allowedActions.includes('Edit Content')) {  // The user has permission to log in to the CMS
+      res.send({ userId, groupIds, accessToken }).status(200);
+    } else {
+      res.statusMessage = "ERROR: User does not have permission to edit website content.";
+      res.status(403).end();
+    }            
+  } catch(err) {
+    console.log(err);
+    console.log(`${err.status}: ${err.msg}`);
+    res.sendStatus(err.status || 400);
+  }
+});
+
+router.post('/groups', (req, res) => {
+  const {userId, groupIds} = req.query;
+  const groupIdsArray = groupIds.split(',');
+
+  // Check that the access token was sent along in the request
+  const authHeader = req.get("Authorization");
+  if (!authHeader) {
+    res.status(401).set('WWW-Authenticate', 'Bearer').send('No auth header found.').end();
+    return;
+  }
+
+  const [type, accessToken] = authHeader.split(' ');
+  if (type !== "Bearer" || !accessToken) {
+    res.status(401).set('WWW-Authenticate', 'Bearer').send('No bearer token found.').end();
+    return;
+  }
+
+  db.featurePermissions.
+    updateUserGroups(userId, groupIdsArray, accessToken)
+    .then((resp) => {
+      res.send(resp).status(200);
+    }).catch(err => {
+      console.log("Error: Failed to sync group membership info. with data from Google Groups.");
+      res.sendStatus(err.status || 400);
+    });
+});
+
+router.get('/picture/:userId', (req, res) => {
+  const { userId } = req.params;
+  db.users.getProfilePicture(userId).then((picture) =>
+    res.send({
+      picture,
+    }),
+  );
+});
+
+// isLead takes a user's email and returns true/false whether they are a 'Lead'
+async function isLead(userEmail) {
+
+  // code borrowed from teamhub `google-auth.js`
+  // we will need to replace the teamhubbackend service account with
+  //  an executive-dashboard service account once we find someone who
```

Comment:
@jmawloo , @sashco-m , Emrys can do this. I'm okay leaving it this way for the MVP.

We can bother Emrys about this after the final teamwide :)

Reply:
Sounds good to me :)